forum_leaderboard/forum_leaderboard.py

```diff
@@ -65,12 +88,19 @@ def student_view(self, context=None):
         # crash. Force call of the author view.
         if getattr(self.runtime, 'is_author_mode', False):
             return self.author_view()
-        course = self.location.course_key
+
+        course = self.get_course()
         threads = cc.Thread.search({
             'course_id': unicode(course), 'commentable_id': self.discussion_id,
             'sort_key': 'votes', 'per_page': self.count
```

Comment:
Apparently `per_page` is weird and actually needs to be given `self.count - 1` or else this thing returns one more thread than you would expect - unless I'm missing something.